docs/platforms/apple/common/configuration/app-hangs.mdx

```diff
@@ -18,6 +18,38 @@ The SDK reports an app hang immediately, but doesn’t report the exact duration
 
 Because the app hang detection integration uses SentryCrashIntegration to capture the stack trace when creating app hang events, SentryCrashIntegration has to be enabled for the integration to work.
 
+You can filter app hang events in beforeSend by checking the event exception type:
+
+<SignInNote />
+
+```swift {tabTitle:Swift}
+import Sentry
+
+SentrySDK.start { options in
+    options.dsn = "___PUBLIC_DSN___"
+    options.beforeSend = { event in
+        if (event.exceptions?.first?.type == "App Hanging") {
```

Comment:
`l`: is this preferred to checking mechanism , e.g `if event.exceptions?.contains(where: { $0.mechanism?.type = "AppHang" }) {`?

Reply:
Right now yes. We could also consider making isAppHangEvent of the Event.m public https://github.com/getsentry/sentry-cocoa/blob/aa45f3606be62823e6f0fd8f0d6a704b16867e17/Sources/Sentry/SentryEvent.m#L203-L207.